source/frontend/src/features/leases/add/AddLeaseContainer.tsx

```diff
@@ -70,7 +70,17 @@ export const AddLeaseContainer: React.FunctionComponent<
     }
   };
 
-  const handleSave = () => {
+  const handleSave = (error?: string | undefined) => {
```

Comment:
Can you explain why passing the error in the params is necessary? You should be able to just use the errorMessage from the state above. Actually, not sure you need this check at all, setting the errorMessage to 'Please check the required fields.' when it is already set to that value should not result in a re-render.

Reply:
Your right, passing it in as a param is unnecessary so I will change that along with the check too as I do see now that if the value of errorMessage is not changing then it wont be re-rendered.